Tests/SourceKitLSPTests/SwiftInterfaceTests.swift

```diff
@@ -0,0 +1,154 @@
+    XCTAssertTrue(fileContents.hasPrefix("import "))
+  }
+  
+  func testOpenInterface() throws {
+    guard let ws = try staticSourceKitSwiftPMWorkspace(name: "SwiftPMPackage") else { return }
+    try ws.buildAndIndex()
+    let call = ws.testLoc("Lib.foo:call")
+    try ws.openDocument(call.url, language: .swift)
+    let workspace = try XCTUnwrap(ws.testServer.server?.queue.sync {
+      try XCTUnwrap(ws.testServer.server?.workspaceForDocument(uri: call.docUri))
+    })
+    let swiftLangServer = try XCTUnwrap(ws.testServer.server?._languageService(for: call.docUri,
+                                                                               .swift,
+                                                                               in: workspace))
+    let expectation = expectation(description: "open interface request")
+    let openInterface = OpenInterfaceRequest(textDocument: call.docIdentifier, name: "lib")
+    let request = Request(openInterface, id: .number(1), clientID: ObjectIdentifier(swiftLangServer),
+                          cancellation: CancellationToken(), reply: { (result: Result<OpenInterfaceRequest.Response, ResponseError>) in
+      do {
+        let interfaceDetails = try result.get()
+        XCTAssertTrue(interfaceDetails.uri.pseudoPath.hasSuffix("/lib.swiftinterface"))
+        let fileContents = try XCTUnwrap(interfaceDetails.uri.fileURL.flatMap({ try String(contentsOf: $0, encoding: .utf8) }))
+        XCTAssertTrue(fileContents.contains("""
+          public struct Lib {
+          
+              public func foo()
+          
+              public init()
+          }
+          """))
+      } catch {
+        XCTFail(error.localizedDescription)
+      }
+      expectation.fulfill()
+    })
+    
+    // Send an arbitrary request through the front door first or SourceKitServer won't be properly initialized. 
+    _ = try ws.sk.sendSync(HoverRequest(
+      textDocument: call.docIdentifier,
+      position: Position(line: 0, utf16index: 8)))
+    swiftLangServer.openInterface(request)
+    
+    waitForExpectations(timeout: 15)
+  }
+
+  func testSwiftInterfaceAcrossModules() throws {
+    guard let ws = try staticSourceKitSwiftPMWorkspace(name: "SwiftPMPackage") else { return }
+    try ws.buildAndIndex()
+    let call = ws.testLoc("Lib.foo:call")
+    try ws.openDocument(call.url, language: .swift)
+    let _resp = try withExtendedLifetime(ws) {
+      try ws.sk.sendSync(DefinitionRequest(
+        textDocument: call.docIdentifier,
+        position: Position(line: 0, utf16index: 8)))
+    }
+    let resp = try XCTUnwrap(_resp)
+    guard case .locations(let locations) = resp else {
+      XCTFail("Unexpected response: \(resp)")
+      return
+    }
+    XCTAssertEqual(locations.count, 1)
+    let location = try XCTUnwrap(locations.first)
+    XCTAssertTrue(location.uri.pseudoPath.hasSuffix("/lib.swiftinterface"))
```

Comment:
Shouldn’t we be jumping to the original source file if it’s in the same package instead of to the Swift interface. AFAICT that’s what we are doing right now. Jumping to the Swift interface here seems like a regression to me.

Reply:
It jumps to the Swift interface only if the symbol is a module. Note the `position` in the request is on an import statement. Otherwise it goes to the appropriate line in the source file.

Reply:
Oh, I didn’t read carefully enough and was assuming that you are jumping to definition on the call. Instead of having the slightly fragile hardcoded position in this test case, the intended design would be to modify `SwiftPMPackage/Sources/exec/main.swift` to

```swift
import /*lib:import*/lib

Lib() . /*Lib.foo:call*/foo()
```

and then you can use define
```diff
- let call = ws.testLoc("Lib.foo:call")
+ let importedModule = ws.testLoc("lib:import")
```
and use `importedModule.position` instead of `Position(line: 0, utf16index: 8)`.

Same for the other test cases in this file.